shell.nix

```diff
@@ -0,0 +1,11 @@
+{ withHoogle ? false
+}:
+let
+  pkgs  = import ./.;
+in
+with builtins; with pkgs.compiler;
+  shellFor {
```

Comment:
Nix has so many things you have to "just know"

Reply:
It's a fairly recent feature, yes.. https://github.com/NixOS/nixpkgs/pull/36393

Reply:
..and not documented in https://nixos.org/nixpkgs/manual/#users-guide-to-the-haskell-infrastructure

Reply:
Exactly :) I only just found that page yesterday. It would have been so helpful in the past.